src/main/java/frc/utils/controllers/keyboard/KeyboardController.java

```diff
@@ -0,0 +1,152 @@
+        this.ZERO = new Trigger(new LoggedDashboardBoolean("Keyboard/0", false)::get);
+        this.MINUS = new Trigger(new LoggedDashboardBoolean("Keyboard/-", false)::get);
+        this.EQUALS = new Trigger(new LoggedDashboardBoolean("Keyboard/=", false)::get);
+        this.BACKSPACE = new Trigger(new LoggedDashboardBoolean("Keyboard/backspace", false)::get);
+        this.TAB = new Trigger(new LoggedDashboardBoolean("Keyboard/tab", false)::get);
+        this.Q = new Trigger(new LoggedDashboardBoolean("Keyboard/q", false)::get);
+        this.W = new Trigger(new LoggedDashboardBoolean("Keyboard/w", false)::get);
+        this.E = new Trigger(new LoggedDashboardBoolean("Keyboard/e", false)::get);
+        this.R = new Trigger(new LoggedDashboardBoolean("Keyboard/r", false)::get);
+        this.T = new Trigger(new LoggedDashboardBoolean("Keyboard/t", false)::get);
+        this.Y = new Trigger(new LoggedDashboardBoolean("Keyboard/y", false)::get);
+        this.U = new Trigger(new LoggedDashboardBoolean("Keyboard/u", false)::get);
+        this.I = new Trigger(new LoggedDashboardBoolean("Keyboard/i", false)::get);
+        this.O = new Trigger(new LoggedDashboardBoolean("Keyboard/o", false)::get);
+        this.P = new Trigger(new LoggedDashboardBoolean("Keyboard/p", false)::get);
+        this.A = new Trigger(new LoggedDashboardBoolean("Keyboard/a", false)::get);
+        this.S = new Trigger(new LoggedDashboardBoolean("Keyboard/s", false)::get);
+        this.D = new Trigger(new LoggedDashboardBoolean("Keyboard/d", false)::get);
+        this.F = new Trigger(new LoggedDashboardBoolean("Keyboard/f", false)::get);
+        this.G = new Trigger(new LoggedDashboardBoolean("Keyboard/g", false)::get);
+        this.H = new Trigger(new LoggedDashboardBoolean("Keyboard/h", false)::get);
+        this.J = new Trigger(new LoggedDashboardBoolean("Keyboard/j", false)::get);
+        this.K = new Trigger(new LoggedDashboardBoolean("Keyboard/k", false)::get);
+        this.L = new Trigger(new LoggedDashboardBoolean("Keyboard/l", false)::get);
+        this.SEMICOLON = new Trigger(new LoggedDashboardBoolean("Keyboard/;", false)::get);
+        this.APOSTROPHE = new Trigger(new LoggedDashboardBoolean("Keyboard/'", false)::get);
+        this.LEFT_SHIFT = new Trigger(new LoggedDashboardBoolean("Keyboard/shift", false)::get);
+        this.Z = new Trigger(new LoggedDashboardBoolean("Keyboard/z", false)::get);
+        this.X = new Trigger(new LoggedDashboardBoolean("Keyboard/x", false)::get);
+        this.C = new Trigger(new LoggedDashboardBoolean("Keyboard/c", false)::get);
+        this.V = new Trigger(new LoggedDashboardBoolean("Keyboard/v", false)::get);
+        this.B = new Trigger(new LoggedDashboardBoolean("Keyboard/b", false)::get);
+        this.N = new Trigger(new LoggedDashboardBoolean("Keyboard/n", false)::get);
+        this.M = new Trigger(new LoggedDashboardBoolean("Keyboard/m", false)::get);
```

Comment:
Recognize that all letters together please, maybe do line space between blocks (letters, f's, numpad, arrows...) to make it easier for the eyes.

Reply:
no, its by place on the keyboard

Reply:
But it's harder to read

Reply:
ok